core/arch/arm/plat-imx/imx_pl310.c

```diff
@@ -68,13 +68,6 @@ void arm_cl2_enable(vaddr_t pl310_base)
 
 vaddr_t pl310_base(void)
 {
-	static void *va;
-
-	if (cpu_mmu_enabled()) {
-		if (!va)
-			va = phys_to_virt(PL310_BASE, MEM_AREA_IO_SEC);
-		return (vaddr_t)va;
-	}
-	return PL310_BASE;
+	return core_mmu_get_io_va(PL310_BASE);
```

Comment:
I could not follow you about using memtype and assert pl310 is secure. the phys_to_virt_io will find the map in secure mapping, then non-secure if not found in secure.